### ISSUE TITLE
Wrong error message: `SyntaxError: Cannot read property 'type' of undefined (0:0)`

### ISSUE THREAD

**Issue by ups-ishii** [prettier/prettier#9226]
**Environments:**

- Prettier Version: 2.1.2
- Running Prettier via: CLI
- Runtime: Node.js v12.16.1
- Operating System: macOS

**Steps to reproduce:**

```shell
cat <<_EOF > sample.js
const p = {
  set name(){}
};
_EOF
npx prettier@2.1.2 sample.js
```

**Expected behavior:**

`SyntaxError: setter must have exactly one formal parameter (2:3)` is expected.

* `npx prettier@1.18.2 sample.js`

```
sample.js[error] sample.js: SyntaxError: setter must have exactly one formal parameter (2:3)
[error]   1 | const p = {
[error] > 2 |   set name(){}
[error]     |   ^
[error]   3 | };
[error]   4 |
```

**Actual behavior:**

* `npx prettier@2.1.2 sample.js`

```
sample.js[error] sample.js: SyntaxError: Cannot read property 'type' of undefined (0:0)
[error]   1 | const p = {
[error]   2 |   set name(){}
[error]   3 | };
```

**FYI**
It looks babel/parser throws right error but Prettier fails to parse the babel/parser error. may be.

* `npx @babel/parser sample.js`
```
setter must have exactly one formal parameter (2:2)
```

Thank you

**Comment by fisker** [comment]
Upstream bug, tracked https://github.com/babel/babel/issues/12074

**Comment by fisker** [comment]
This is fixed by #9408

**Comment by ups-ishii** [comment]
Thank you very mush!
